src/game/SpellAuras.cpp

```diff
@@ -4389,6 +4389,33 @@ void Aura::PeriodicTick()
                     }
                 }
             }
+            
+            // Hunter's Improved Mend Pet dispel check
+            if (spellProto->SpellFamilyName == SPELLFAMILY_HUNTER && spellProto->SpellFamilyFlags && spellProto->SpellIconID == 267)
+            {
+                // Check caster has one of the ranks of Improved Mend Pet
+                if (pCaster->HasSpell(19572) || pCaster->HasSpell(19573)) {
+                    
+                    // The first tick is fired when channel starts, so only check other ticks
+                    if (GetAuraTicks() > 1)
+                    {
+                        int32 triggerAmount = 0;
+                        
+                        // Set threshold for dispel check
+                        if (pCaster->HasSpell(19572))
+                            triggerAmount = 15; //rank 1
+                            
+                        if (pCaster->HasSpell(19573))
+                            triggerAmount = 50; //rank 2
+
```

Comment:
Is there a better way to get the triggerAmount? I tried to do it via Talents, or spellProto modifiers, but could never return a value.